src/net.cpp

```diff
@@ -1467,8 +1467,6 @@ void CConnman::SocketHandler()
         }
     }
 
-    NodesSnapshot snap{*this};
```

Comment:
157674ef213e6329215d1b7eb85dacbc23c468db

This snap catches nodes added in the above `AcceptConnection` where the new snap at the beginning doesn't. Is this intentional?

Reply:
Hmm, good catch! No, that is not intentional.

So, the previous code did:

1. `SocketEvents()`, adding ready sockets from `vNodes` to `{recv,send,error}_set`.
2. `AcceptConnection()`, adding new entries to `vNodes`.
3. Loop over `vNodes`, for each node:
3.1. Check whether its socket is in one of `{recv,send,error}_set`. Observation - the sockets of nodes that were added in 2. will not be in any of `{recv,send,error}_set`, so for them all of `recvSet`, `sendSet` and `errorSet` will be `false`.
3.2. `InactivityCheck()` - will also be done for nodes added in 2.

The new code will skip nodes added in 2. in the 3. loop. That's the same as the old code with respect to 3.1. The only difference is that the new code will not do `InactivityCheck()` for newly accepted nodes. I think that is ok because it is unlikely that a newly accepted peer would be considered inactive. In the rare case where this might have happened in the old code and will not happen in the new code, the new code will do the check on the next call to `SocketHandler()`.

What do you think? Leave it as is or somehow ensure that new and old behave in an identical way?

Reply:
if `InactivityCheck` for new nodes is pointless then maybe move `// Accept new connections` to the end?

Reply:
Added an explicit commit (a71b3f426) to show the intention. Putting the accept at the end somehow looked strange to me.

Maybe these two actions deserve to be split in separate functions (ie split `SocketHandler()`): 1. checking which of the listening sockets are ready for IO and accepting connections on them and 2. checking which of the already connected sockets are ready for IO and processing them.

Reply:
Since you are not splitting maybe add a comment for now like `// Take a snapshot of current nodes before acceptation new ones - new nodes are irrelevant below` or something like that.

Reply:
Right, a comment is warranted. Added:

```cpp
    // Accept new connections. Done after taking a snapshot of the nodes because
    // `AcceptConnection()` will add new entries to `vNodes` which are unwanted
    // in the loop below because:
    // - all of `recvSet`, `sendSet` and `errorSet` will be `false` for them
    // - `InactivityCheck()` will either be a noop or will wrongly command to
    //   disconnect the node before we have tried sending or receiving anything
    //   to it.
```